### ISSUE TITLE
S4143: False positive when increment operator used in index

### ISSUE THREAD

**Issue by salixzs** [SonarSource/sonar-dotnet#2038]
### Description

When increment operator (`variable++`) used in series of array assignments, it is falsely reported as S4143.

### Repro steps

```csharp
string[] formatVariations = new string[5];
int counter = 0;
formatVariations[counter++] = pattern1;
formatVariations[counter++] = pattern2;
formatVariations[counter++] = pattern3;
formatVariations[counter++] = pattern4;
formatVariations[counter++] = pattern5;
```

### Expected behavior

It is not reported as S4143.

### Actual behavior

S4143 is reported, with message
`Verify this is the index/key that was intended; a value has already been set for it.`

### Known workarounds

None

### Related information

* SonarC# Version = latest
* Visual Studio Version = 2017
* If running through the Scanner for MSBuild, its version 15

**Comment by valhristov** [comment]
I managed to reproduce this problem with SonarC# 7.5, but it was fixed with 7.6 and is not reproducible with 7.9 (the current latest version):
https://github.com/SonarSource/sonar-dotnet/releases/tag/7.6.0.7024

I do get an issue from S1854 on the last line of the assignments. I think it is not a false positive though, because the new incremented value is indeed unused.
